utils/keys_utils.go

```diff
@@ -0,0 +1,166 @@
+package utils
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+const (
+	KeysPathKey = "KEYS_PATH"
+)
+
+func Gen(id string) {
+	keyspath := os.Getenv(KeysPathKey)
+	savePrivateFileTo := keyspath + "/" + id + ".priv"
+	savePublicFileTo := keyspath + "/" + id + ".pub"
+	bitSize := 4096
+
+	privateKey, err := GeneratePrivateKey(bitSize)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	privateKeyBytes, publicKeyBytes := encodeKeysToPem(privateKey, &privateKey.PublicKey)
+
+	err = writeKeyToFile(privateKeyBytes, savePrivateFileTo)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	err = writeKeyToFile([]byte(publicKeyBytes), savePublicFileTo)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func GetPrivateKey(id string) *rsa.PrivateKey {
+	keyspath := os.Getenv(KeysPathKey)
+	readPrivKey, err := ioutil.ReadFile(keyspath + "/" + id + ".priv")
+
+	if err != nil {
+		log.Fatal("The private key is not where it should be")
+	}
+
+	pemDecodedPrivKey, rest := pem.Decode(readPrivKey)
+
+	if len(rest) > 0 {
+		log.Fatal("Error on decoding private key; the rest is not empty.")
+	}
+
+	rsaPrivateKey, err := x509.ParsePKCS1PrivateKey(pemDecodedPrivKey.Bytes)
+	if err != nil {
+		log.Fatal("Error on parsing private key " + err.Error())
+	}
+
+	return rsaPrivateKey
+}
+
+func SignMessage(privateKey *rsa.PrivateKey, message []byte) ([]byte, []byte) {
+	messageHash := sha256.New()
+	writtenBytesCounter, err := messageHash.Write(message)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if writtenBytesCounter != len(message) {
+		log.Fatal("The message has not been entirely written in the message hash.")
+	}
+
+	msgHashSum := messageHash.Sum(nil)
+
+	signature, err := rsa.SignPSS(rand.Reader, privateKey, crypto.SHA256, msgHashSum, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	return signature, msgHashSum
+}
+
+func VerifySignature(key *rsa.PublicKey, hash []byte, signature []byte) bool {
+	err := rsa.VerifyPSS(key, crypto.SHA256, hash, signature, nil)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func GetPublicKey(id string) *rsa.PublicKey {
```

Comment:
@raonismaneoto PTAL :)

Reply:
@raonismaneoto I think you missed this comment

Reply:
actually I've refactored them, but not this line. Check the whole change please. I think if we try to merge both functions it will lead to some additional conditions, besides the return type will be confused, once the func must handle both private and public key.

Reply:
@paulofelipefeitosa , could you check it please?

Reply:
Hi @raonismaneoto you're right! There is no problem with that! Thanks.